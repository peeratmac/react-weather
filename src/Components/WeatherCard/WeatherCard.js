import React, { Component } from 'react';
import { connect } from 'react-redux';
import './WeatherCard.css';
import { fetchUsingStationID } from '../../apiCalls';

class WeatherCard extends Component {
  render() {
    console.log('WeatherCard props', this.props);
    const { cities } = this.props;

    return cities.map((city, i) => {
      return (
        <div className='card'>
          <section onClick={event => console.log(event.target)}>
            <h1 key={i}>{city.title}</h1>
            <h3>WeatherStationID: {city.woeid}</h3>
            <h3>Lat/Long: {city.latt_long}</h3>
            {/* <h4>{fetchUsingStationID(city.woeid)}</h4> */}
          </section>
        </div>
      );
    });
  }
}

const mapStateToProps = state => ({
  cities: state.cities
});

export default connect(
  mapStateToProps,
  null
)(WeatherCard);
