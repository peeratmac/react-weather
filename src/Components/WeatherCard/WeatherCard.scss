* {
  box-sizing: border-box;
  font-family: 'Rubik', sans-serif;
}

$base-weather: #38ada9;
$city-label: #82ccdd;
$weather-station-id-label: #45aaf2;
$weather-station-id: #f8c291;
$lat-long-label: #78e08f;
$lat-long: #1dd1a1;
$red-alert-error: #ee5253;
$newly-append-card: #ff9f43;

.card {
  border: 10px solid $base-weather;
  border-radius: 25px;
  margin: 10px;
  text-align: center;
  padding: 20px;
  background: url('https://www.metaweather.com/static/img/weather/lc.svg');
}

.city-title {
  background-color: $city-label;
  padding: 5px;
  border-radius: 5px;
}

.weather-station-label,
.current-condition-label {
  background-color: $weather-station-id-label;
  padding: 5px;
  border-radius: 5px;
}

.weather-station-id {
  background-color: $weather-station-id;
  padding: 5px;
  border-radius: 5px;
}

.lat-long-label {
  background-color: $lat-long-label;
  padding: 5px;
  border-radius: 5px;
}

.lat-long {
  background-color: $lat-long;
  padding: 5px;
  border-radius: 5px;
}
