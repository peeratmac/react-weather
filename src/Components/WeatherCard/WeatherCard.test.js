import React from 'react';
import { shallow } from 'enzyme';
import { WeatherCard } from './WeatherCard';

describe('WEATHER CARD', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WeatherCard
        stationIDs={[57980]}
        weatherInfo={[]}
        currentStation={47889}
        cities={[]}
      />
    );
  });

  it('should match the WeatherCard Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
