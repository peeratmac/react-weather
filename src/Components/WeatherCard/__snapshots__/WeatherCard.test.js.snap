// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WEATHER CARD should match the WeatherCard Snapshot 1`] = `null`;
